cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -215,32 +230,35 @@ def setup_leader_authentication():
     # eg:
     # {'/root/cdk/serviceaccount.key': 'RSA:2471731...'}
     charms.leadership.leader_set(leader_data)
-
     set_state('authentication.setup')
 
 
 @when_not('leadership.is_leader')
-@when_not('authentication.setup')
 def setup_non_leader_authentication():
```

Comment:
That's a good catch! The password update flow should should go like this:
- User sets new passowrd
- Leader grabs the password and propagates to non-leaders
- Leader restarts
- Non-leaders get the new password and restart.

You will notice I added the 'leadership.is_leader' on the `password_changed()` so that only the leader acts on password change.

To your question, I think the change of leadership data will trigger the reactive framework on the non-leaders in a timely fashion so there will not be the 5 minute wait.